crates/sats/src/relation.rs

```diff
@@ -175,18 +179,139 @@ pub struct HeaderOnlyField<'a> {
     pub fields: Vec<ColumnOnlyField<'a>>,
 }
 
+/// Represents a collection of `fields` metadata made of [Column].
+///
+/// The `fields` need to be preserved by the order specified by the user.
+#[derive(Debug, Clone, Eq, PartialOrd, Ord, Hash)]
+pub struct Fields {
+    columns: Vec<Column>,
+    /// Keeps an index to quickly look up the position
+    idx: BTreeMap<Arc<FieldName>, usize>,
```

Comment:
Again, I actually don't think `Arc` is necessary here. Also I don't think we should use FieldName here but rather
```rust
idx: BTreeMap<(String, TableId), usize)>
```

I will explain below.